### ISSUE TITLE
Page Editor Rendering Issues - Orderables/InlinePanels, with Comments, etc.

### ISSUE THREAD

**Issue by dkirkham** [wagtail/wagtail#7971]
### Issue Summary

The Page Editor UI has a number of rendering issues when there are multiple editor panels and when comments are activated, especially with InlinePanels/Orderables.

This issue has been lodged to assist with development of the new page editor, and are probably in scope for attention in that development. See discussion at #7739.

### Common Steps to Reproduce

1. Start a fresh Wagtail installation as per the instructions at [Getting Started](https://docs.wagtail.org/en/stable/getting_started/tutorial.html).
2. Merge in the application at https://github.com/dkirkham/wagtail-ui-testpage
3. Add `'testpage'` to `INSTALLED_APPS` in `settings/base.py`
4. Migrate and runserver
5. Login to the Wagtail admin interface
6. Create a child `Testpage` of the home page
7. Further steps as described below

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

Note that the testpage application is based on a production application, but with all the view and other business logic removed.

### Technical details

* Wagtail/Django versions: 2.13/3.2 and 2.16/4.0 (and likely the intermediate versions)
* Browser versions: MacOS Firefox (ca. v96), Chrome (ca. v98), Safari(v14)
  * Also iPad Pro in landscape and portrait exhibit similar issues to those described below for a desktop browser.
* Python version: probably irrelevant, but I've observed with 3.7 and 3.8

### Comments Toggle overlaying Editor Panel tabs

To reproduce (from editing a new `Test page` above):
1. Set browser width to approx 1230px
2. Slowly reduce the width of the browser
3. Observe the Comments Toggle overlaying the Settings tab

![image](https://user-images.githubusercontent.com/1977376/153682561-1708a662-77f7-449f-9384-3d66f1a96775.png)

### Comments Toggle changing vertical location

To reproduce (from above state)
1. Continue to slowly reduce the browser width
2. (Wagtail 2.13) Observe the empty space that appears under the Editor Panel tabs and that the Comments Toggle is in that space
3. (Wagtail 2.16) Observe the Comments Toggle moves downwards, partially obscured

![image](https://user-images.githubusercontent.com/1977376/153682696-97c69c22-eaa9-4024-a3ff-c26a4ffc92fa.png)
![image](https://user-images.githubusercontent.com/1977376/153682780-2259943f-e066-496b-8c98-0c8424b051b5.png)

### Editor Panel tabs wrapping untidily

To reproduce (from above state)
1. Continue to slowly reduce the browser width
2. Observe the `Settings` tab moves to the second row, not left aligned with the `Title & Introduction` tab
3. As the browser width is further reduced, the Comments Toggle jumps up and down, and the tabs continue to stack in two rows

![image](https://user-images.githubusercontent.com/1977376/153682995-f6d2d78a-f012-4403-9c72-e7db7ac8e74f.png)

### Activated Comments button under field help text

To reproduce:
1. Return browser width to about 1230px
2. Go to the `Title & Introduction` tab
3. Hover the pointer over each of the Title, Time field and Hero Image fields
4. Observe that the help text alignment is uneven
5. Observe that the comment button is probably under the help text of the time field and inoperable (depends on exact browser width)

![image](https://user-images.githubusercontent.com/1977376/153683073-b09d2e15-15c0-476d-82c9-cad72355b702.png)
[Note this is a composite image - only one help text will appear at a time]

### InlinePanel/Orderable Layout with/without Comments activated

To reproduce (from editing a new `Test page`):
1. Return browser width to about 1230px
2. Go to the `Event Details` tab
3. Add an `Orderable`
4. Observe the layout is reasonable, but the field labels could have a bit more width, and less space on the right
5. Activate comments
6. Observe the InlinePanel is even more narrow, not enough horizontal space for the field labels, the fields themselves are excessively narrow, and lots of wasted space on the right.

![image](https://user-images.githubusercontent.com/1977376/153683258-581aed81-0403-48cf-86cf-0dc89969b9b5.png)
![image](https://user-images.githubusercontent.com/1977376/153683494-15157e55-0af1-4d8c-b227-0237724c14d6.png)

### Scroll stops when pointing at a Time Field in an InlinePanel - Firefox

To reproduce (from editing a new `Test page`):
1. Go to the `Event Details` tab, scrolled to the top
2. Add an `Orderable`
3. On a touch pad device (eg. Mac Air), point on the `Sub page` field
4. Scroll down
5. Observe that the scroll stops when the pointer is on the Time field, but does not stop at the other field types (even the datetime field)

Note: I haven't tested this with a wheel mouse.

### Scroll stops when pointing at a Time Field in an InlinePanel - Safari, Chrome

To reproduce (from editing a new `Test page`):
1. Go to the `Event Details` tab, scrolled to the top
2. Add an `Orderable`
3. Point at the `Time field` such that it has recognised the hover, and try to scroll
4. Observe that the page does not scroll

Note: I haven't tested this with a wheel mouse.

### Unsaved Edits indicator doesn't show on Orderable reordering

To reproduce:
1. Create a Test Sub Page as a child of the Homepage with random title, and Publish
2. Create a Test Page as a child of the Homepage with random field contents, including any random image upload, and at least three instances of the `Orderable` on the `Event Details` tab; then Publish. (Don't bother to preview, I haven't created a template).
3. Return to the admin page navigator, and Edit the newly created Testpage
4. Go to the `Event Details` tab
5. Reorder the `Orderables` with the up/down arrow buttons; make multiple changes
6. Observe that the Unsaved Edits notification _does not appear_
7. Observe that an attempt to leave the page is not detected

I note that it appears that multiple changes are required to fields before the Unsaved Edits notification appears, and that perhaps there is some sort of threshold that must be reached before the notification is generated.

**Comment by dkirkham** [comment]
I've just reassessed the UI issues described above with 3.0rc3 to see which have been addressed with the page editor updates and which are still causing issues. I fully appreciate not all of these issues have specifically addressed in the first tranche of page editor changes.

In summary: a few issues have been resolved, albeit a couple of those now have other UI issues. Other issues have not yet been addressed.

### 1. Comments Toggle overlaying Editor Panel tabs

Mostly solved, with comment switch spills over to a new row:

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/1977376/168426722-acdea616-b7b6-4c13-850a-ef4e4a744202.png">

However, the "Comments notifications" text in the example above is not aligned with the first tab ("Title & Introduction"), which it is when the screen is made more narrow (<800px):

<img width="798" alt="image" src="https://user-images.githubusercontent.com/1977376/168426805-794f1b84-6cd9-470d-8dee-375d2ba8c91f.png">

### 2. Comments Toggle changing vertical location

Appears solved.

### 3. Editor Panel tabs wrapping untidily

Effectively solved by making that part of the page horizontally scrollable.

### 4. Activated Comments button under field help text

Still occurs but it is harder to find browser widths where there is overlap.
Field help text is still badly aligned when comments are enabled.

<img width="1146" alt="image" src="https://user-images.githubusercontent.com/1977376/168427251-64012432-58ff-4a17-8b9a-bb1475a5ccc4.png">
<img width="1146" alt="image" src="https://user-images.githubusercontent.com/1977376/168427269-816f8beb-1166-4909-8e31-9712d9981b2c.png">
<img width="1146" alt="image" src="https://user-images.githubusercontent.com/1977376/168427278-04cf2d55-a017-4b82-b57b-6878daf937ae.png">

### 5. InlinePanel/Orderable Layout with/without Comments activated

Appears not to have been addressed so far.
Comment icons are not always appearing in the Orderable/InlinePanel fields.

### 6. Scroll stops when pointing at a Time Field in an InlinePanel - Firefox

Appears not to have been addressed so far.

### 7. Scroll stops when pointing at a Time Field in an InlinePanel - Safari, Chrome

Appears not to have been addressed so far.

### 8. Unsaved Edits indicator doesn't show on Orderable reordering

Appears not to have been addressed so far.

### 9. An new issue - activated comments result in field buttons being cropped

On the Event Detail tab, in the Orderables/InlineFields:
* the buttons on the first Orderable are cropped by an overly-narrow field space.

<img width="868" alt="image" src="https://user-images.githubusercontent.com/1977376/168429148-c6bc316b-06d0-4f4a-9cb8-3cebb8af4f88.png">

**Comment by thibaudcolas** [comment]
Just wanted to say thank you @dkirkham for taking the time to report this so well. I’m reviewing what our recent changes will / won’t have fixed, and this helps a lot.

**Comment by dkirkham** [comment]
Thanks @thibaudcolas, and apologies for the tardy response - I was travelling in a very remote area with little connectivity most of the last month.

The new editor UI is looking good, and many of the report issues have been resolved. There are new ones, for instance with `MultiFieldPanel` `help_text`, that I've described in #9134.

As part of that issue, I've updated my test repo for 3.0 and 4.0 compatibility. To reproduce this issue in Wagtail 2.x, you'll now need to check out commit `adce6df`.

**Comment by dkirkham** [comment]
I've now upgraded to Wagtail 6.2 and reassessed these layout issues.

One the whole, the new page editor layouts that have emerged during the Wagtail 5 and 6 series have solved or eliminated most of the problems I initially raised.

Relative to my [20 month-old comments](https://github.com/wagtail/wagtail/issues/7971#issuecomment-1311632970) above:

- 1 has been solved with the new comments panel
- 6 & 7 have improved such that it is no longer a problem, at least to me - I can now scroll by the TimeFields in an Orderable and the scroll will not stall like it used to. There are still one or two funny scrolling behaviours going on with the TimeField and Orderable, which I'll describe in a fresh issue - something that might attract a JS developer wanting to test their debugging skills, but not at the expense of other more important bug fixes and development.
  - I also spotted an issue with comments around TimeField and DateTimeFields, which I'll also report separately.
- 8 appears to have been fixed - at least I can't fault it like I could before.
- 9 has been mostly alleviated by using the triple-dot menu, rather than large buttons.

As a result, I'll close this issue as completed, with thanks to everyone who worked on the new page editor layout.
